### ISSUE TITLE
Enhancement: `disableTypeChecked` for JS files by default if `allowJs` but not `checkJs`

### ISSUE THREAD

**Issue by karlhorky** [typescript-eslint/typescript-eslint#9583]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://v8--typescript-eslint.netlify.app/troubleshooting/typed-linting#fixing-the-error

### Description

Projects using `allowJs: true` but `checkJs: false` arguably are not looking for TS type problems to be reported in JavaScript files.

The [Typed Linting -> Fixing the error docs section](https://v8--typescript-eslint.netlify.app/troubleshooting/typed-linting#fixing-the-error) recommends disabling type-checked lint rules for JavaScript files, if they should not be linted:

```ts
import tseslint from 'typescript-eslint';

export default tseslint.config(
  // ... the rest of your config ...
  {
    files: ['**/*.js'],
    extends: [tseslint.configs.disableTypeChecked],
  },
);
```

This is a good start and helpful for projects, but it doesn't take into account the `checkJs: false` setting from the `tsconfig.json` file.

The proposal in this issue is that disabling type checked lint rules should be default behavior in the case that `checkJs: false` and `allowJs: true`

(with a possible exception for lint rules that need type information but do not lead to an experience of "type errors")

### Fail

```TypeScript
// A JS file in a `checkJs: false` file which should not have type-checked lint rule problems
```

### Pass

```TypeScript
// A JS file in a `checkJs: true` file which should have type-checked lint rule problems
```

### Alternatives Considered

Add a docs section that shows how to disable type-checked lint rules in JS files, but re-enable them when `checkJs: true` is set, eg:

```ts
import { readFile } from 'node:fs/promises';
import stripJsonComments from 'strip-json-comments';
import stripJsonTrailingCommas from 'strip-json-trailing-commas';
import tseslint from 'typescript-eslint';

const tsconfig = JSON.parse(
  stripJsonComments(
    stripJsonTrailingCommas(
      await readFile(`${process.cwd()}/tsconfig.json`, 'utf-8'),
    ),
  ),
);

export default tseslint.config(
  // ... the rest of your config ...
  tsconfig.compilerOptions.checkJs ? {} : {
    files: ['**/*.js', '**/*.mjs', '**/*.cjs'],
    extends: [tseslint.configs.disableTypeChecked],
  },
);
```

**Comment by JoshuaKGoldberg** [comment]
> disabling type checked lint rules should be default behavior in the case that `checkJs: false` and `allowJs: true`

🤔 this seems like kind of a niche scenario to me. My advice would be to just not enable the type checked rules for `.js` files in the first place - i.e. #9859. Do a lot of projects hit this case?

> `readFile` ... `tsconfig.compilerOptions.checkJs ? {} : {`

This I don't think we'd want in our docs. It's a lot of runtime logic for something that remains constant in almost all projects, almost all of the time.

---

Let's put this under `evaluating community engagement`. If there are other issues filed in this area, we can link them here to see if a lot of folks hit this.

**Comment by JoshuaKGoldberg** [comment]
This issue has been open for several months and hasn't received any more community interests that I know of. Given that there's an alternate recommendation I think we can close it out now. Thanks though!
